data/data/bootstrap/baremetal/files/usr/local/bin/startironic.sh.template

```diff
@@ -183,19 +183,31 @@ podman run -d --name ironic-inspector-ramdisk-logs \
      --entrypoint /bin/runlogwatch.sh \
      -v $IRONIC_SHARED_VOLUME:/shared:z "${IRONIC_INSPECTOR_IMAGE}"
 
-# Now loop so the service remains active and restart everything should one of the containers exit unexpectedly.
-# The alternative would be RemainAfterExit=yes but then we lose the ability to restart if something crashes.
+set +x
+AUTH_DIR=/opt/metal3/auth
+ironic_url="$(printf 'http://%s:%s@localhost:6385/v1' "$(cat "${AUTH_DIR}/ironic/username")" "$(cat "${AUTH_DIR}/ironic/password")")"
+inspector_url="$(printf 'http://%s:%s@localhost:5050/v1' "$(cat "${AUTH_DIR}/ironic-inspector/username")" "$(cat "${AUTH_DIR}/ironic-inspector/password")")"
+
+while [ "$(curl -s "${ironic_url}/nodes" | jq '.nodes[] | .uuid' | wc -l)" -lt 1 ]; do
+  echo "Waiting for a control plane host to show up in Ironic..."
+  sleep 20
+done
+
 while true; do
-    for name in ironic-api ironic-conductor ironic-inspector ironic-deploy-ramdisk-logs ironic-inspector-ramdisk-logs $dnsmasq_container_name httpd mariadb; do
-        # Note there are two levels of go templating here, the outer braces
-        # are for the templating done in openshift-install, to escape the
-        # templating input to the --format option of podman inspect
-        state=$(podman inspect ${name} --format  {{ "{{.State.Status}}" }})
-        if [[ $state != "running" ]]; then
-            echo "ERROR: Unexpected service status for $name"
-            podman inspect ${name}
-            exit 1
-        fi
-    done
+    # Check if all nodes have been deployed
+    if ! curl -s "${ironic_url}/nodes" | jq '.nodes[] | .provision_state' | grep -v active;
+    then
+      echo "All hosts have been deployed."
+      sleep 30
```

Comment:
Should we remove this arbitrary sleep, and just go directly to the test?

It seems likely that the master-bmh-update takes a few seconds to complete, but it's only waiting for the BMH CRs to appear, not for the metal3 pod etc to start, so that could happen pretty early?

Reply:
Removed

Reply:
The sleep is to make sure once the hosts go 'active' terraform has enough time to notice as well, there's a chance if this script notices the hosts go active, we could teardown Ironic while terraform hasn't seen that yet causing the installer to error.

Reply:
Ack, I'm always nervous when I see hard-coded sleeps like this, but unless there is some way we can detect that the installer got beyond the terraform phase I guess it's reasonable